### ISSUE TITLE
Adding a plain ACL-style bibliography entry to the meta-information page. 

### ISSUE THREAD

**Issue by emjotde** [acl-org/acl-anthology#412]
Hi,
I have a feature request. Would it be possible to add a plain text version of a reference based on the ACL-Latex style to the description of a paper, so that it would look like it would appear in a final ACL paper? That would be very useful for Copy&Paste into webpages or Word documents if one does not use Latex for some obscure reason (that happens).

**Comment by akoehn** [comment]
Sounds like a good idea! Would you be willing to work on it?  The only complicated thing is to find a program that does the bib to txt conversion. The rest will be analogous to mods, which you can see in the makefile.

**Comment by davidweichiang** [comment]
[Pybtex](https://pybtex.org) can do that. I agree that this would be nice.

**Comment by mjpost** [comment]
@emjotde, are you interested in having it look exactly like a citation, or just having a blob of text that you can copy into other documents (for, say, when summarizing papers in a conference). For example, would this be sufficient?

> Sarthak Jain, Byron C. Wallace. *Attention is not Explanation*. NAACL 2019.
> https://aclweb.org/anthology/N19-1357/

(or give an example of what you're after).

**Comment by emjotde** [comment]
Close-enough blob is close enough :)

This is what I produced for a recent blog post:

- Marcin Junczys-Dowmunt. 2018a. Dual Conditional Cross-Entropy Filtering of Noisy Parallel Corpora. Proceedings of the Third Conference on Machine Translation: Shared Task Papers. Belgium, pages 888-895. https://aclweb.org/anthology/papers/W/W18/W18-6478/

**Comment by mbollmann** [comment]
FWIW, I've looked into the pybtex suggestion, and it's a bit fiddly because the API only allows for generation of full bibliographies, not single entries. The upside is that we could plug in the most recent `.bst` file from the ACL style files for some extra niceness.

I won't delve further into this right now, but if anyone wants to give it a go, I'd say either:

1. Instantiate a pybtex engine somewhere in `create_hugo_yaml.py` and run each paper through it (using our generated BibTeX as input, and the `markdown` engine as output), then save that as an additional field in the YAML file.

2. Just add a function `as_markdown()` to complement our existing `as_bibtex()` function on the `Paper` class and write some simple formatting logic yourself.

The latter option is certainly *way* easier.

**Comment by mjpost** [comment]
Putting together a post-EMNLP report, I'm reminded how useful this would be. A simple approach to do this would be to change the title, e.g., from `A Streaming Approach For Efficient Batched Beam Search - ACL Anthology` to something like `A Streaming Approach For Efficient Batched Beam Search (Wallace et al., EMNLP 2020)`. Smart editors will often grab the page title when pasting in links, so this would work automatically in those settings.

If we want an actual text reference, I think we should just roll our own via an `as_markdown()` function and create a "Copy Markdown to Clipboard" button, but this would take a bit more work than the title change.

**Comment by nschneid** [comment]
I like the `as_markdown()` idea. I often want to copy a paper's title/authors into an email but without the huge heading font that I get from copying the title directly, and without linking the author names.

**Comment by emjotde** [comment]
+1 for `as_makdown()`

**Comment by mjpost** [comment]
What do we think of this?

![image](https://user-images.githubusercontent.com/455256/100168108-9f6ebb00-2e8e-11eb-92f9-34f97231d740.png)

Clicking produces:

```
[Simulated multiple reference training improves low-resource machine translation](https://www.aclweb.org/anthology/2020.emnlp-main.7) (Khayrallah et al., 2020)
```

which Markdown renders as

> [Simulated multiple reference training improves low-resource machine translation](https://www.aclweb.org/anthology/2020.emnlp-main.7) (Khayrallah et al., 2020)

**Comment by mjpost** [comment]
(Adding the venue is a bit trickier; we could merge this as a step forward.)

**Comment by nschneid** [comment]
Could mimic https://github.com/zotero/translators/blob/d646b4cd0c1cb7b5f450b9f67bbed16dd605c9f3/ACLWeb.js#L149-L194 for the abbreviated venue

**Comment by mbollmann** [comment]
Generally speaking, I like the idea, but I'm not sure I'd want to use Markdown for it, or at least not as the only option. I'd love it (my standard note-taking app uses Markdown), but not everyone is familiar with it, and if you just paste it in an email it'll look weird. So I feel a plain-text alternative would be nice.

**Comment by mjpost** [comment]
What about borrowing what Semantic Scholar does? If you click "Cite", you get a box with tabs that lets you choose what format. Each is presented in a box, with the text already highlighted.

![image](https://user-images.githubusercontent.com/455256/100557880-c0ae1d80-3279-11eb-967f-797e1c77f6b9.png)

We could replace MLA, APA, etc with "Markdown", "plain text", etc.

**Comment by mbollmann** [comment]
That would be the most flexible solution, yes!

(I guess I should look into the implementation side of this? ;))

**Comment by mjpost** [comment]
Yes, @mbollmann, that would be really great!

**Comment by mbollmann** [comment]
The PR #1375 will add the simple markdown citation suggested by @mjpost; I've also briefly looked into what would be required to render a proper citation according to the official ACL BibTeX style. Since this is a bit more involved, I wanted to put this up here for discussion first.

With [pybtex](https://docs.pybtex.org/) and [pypandoc](https://github.com/bebraw/pypandoc) (also requires a [pandoc](https://pandoc.org/) installation), this snippet will produce a proper Markdown-formatted citation:

```python
from pybtex.bibtex import BibTeXEngine
import pypandoc

engine = BibTeXEngine()
# Convert BibTeX entry to (formatted) LaTeX using official ACL .bst file
s = engine.format_from_string(paper.as_bibtex(), style="/path/to/acl_natbib")
# A bit hacky: remove the boilerplate around the \bibitem, because pandoc doesn't handle it correctly ...
s = s[s.index("\\bibitem"):s.index("\\end{thebibliography}")]
# Convert to markdown
pypandoc.convert_text(s, "md", format="latex").replace("\n", " ").strip()
```

Not quite sure about the overhead during the build process, because pypandoc always invokes the external pandoc binary AFAICT. Also, we'd need to commit the `acl_natbib.bst` somewhere in the repo and keep it updated, if necessary.

Worth it? Thoughts? @mjpost @akoehn

**Comment by mjpost** [comment]
Bottom line, I think this would be a great feature that would add polish and that people would find useful in a lot of scenarios (e.g., internal reports, blog posts), so it's worth doing if you have the time and energy for it at the moment. It does seem like there are a few problems to be solved, though.

I assume we'd be embedding this into the web page, not generating a new file.  I expect invoking that binary 66k times is going to slow things down. The question then is by how much, so I really think we'd need an implementation. It's all done by Github so even if it adds ten minutes, it doesn't seem like a big deal. Though it might be nice to be able to disable it (say for previews), which could add some minor complications. I wonder if it would be easy to parallelize this generation?

The official `acl_natbib.bst` is [here](https://raw.githubusercontent.com/acl-org/ACLPUB/master/templates/latex/acl_natbib.bst), so we could easily `wget` it as part of the build process.

If this gets done, I wonder if we should also use pandoc to covert to RTF, too, which could be pasted into areas like Word / Google Docs / Pages? I think having RTF and MD would cover most use cases.

**Comment by akoehn** [comment]
My opinions:
 - we should pin the acl_natbib to a specific version because otherwise our builds are not reproducible anymore
 -  It would be great to make it fast, as we otherwise 1) waste resources and 2) the workflow gets worse (e.g. getting feedback on whether a build was successful)

We only have proceedings and inproceedings, right? Wouldn't it be fairly simple to use a string template?  I know that this might lead to diversion, but OTOH the pybtex approach looks overly complicated in terms of processes run etc.

**Comment by mbollmann** [comment]
Pybtex also has a pure Python rendering engine; if we used that to (re-)create an ACL-style template, we can generate styled Markdown directly from that, without going through pandoc.

**Comment by mjpost** [comment]
Easy to grab a specific acl_natib commit via a similar URL. 

We also have @article and should soon (#918) have @misc. But I agree we could probably roll our own, though I wonder if there are corner cases that natbib solves that we'd later encounter.

**Comment by mjpost** [comment]
Closed via #1390, which added the feature.
